components/bl-credit-card-component/src/index.js

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { CardForm } from './card-form';
+import { CardPreview } from './card-preview';
+import {
+  clearCardForm, ensureMeasure, formatCVC, getCardByNumber, validateCardCVC, validateCardExpiry, validateCardNumber,
+} from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function CreditCardComponent({ component, eventHandlers }) {
+  const {
+    display, classList, style, direction, borderWidth, borderStyle, borderColor, cardPreviewVisibility,
+  } = component;
+
+  const [cardNumber, setCardNumber] = useState('');
+  const [expiry, setExpiry] = useState('');
+  const [cvc, setCVC] = useState('');
+  const [cardholderName, setCardholderName] = useState('');
+  const [focus, setFocus] = useState('');
+  const [card, setCard] = useState();
+
+  const creditCardRef = useRef(null);
+
+  useEffect(() => {
+    component.el = creditCardRef.current;
+  }, []);
+
+  useEffect(() => {
+    const card = getCardByNumber(cardNumber);
+
+    setCard(card);
+  }, [cardNumber]);
+
+  useEffect(() => {
+    if (cvc) {
+      const value = formatCVC(cvc, card);
+
+      setCVC(value);
+    }
+  }, [card]);
+
+  const styles = {
+    flexDirection: direction,
+    borderWidth  : ensureMeasure('border-width', borderWidth),
+    borderStyle,
+    borderColor,
+    ...style,
+  };
+
+  Object.assign(component, {
+    clearForm     : () => clearCardForm(setCardNumber, setExpiry, setCVC, setCardholderName),
+    validateNumber: () => validateCardNumber(cardNumber, card),
+    validateExpiry: () => validateCardExpiry(expiry),
+    validateCVC   : () => validateCardCVC(cvc, card),
+  });
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div ref={ creditCardRef } className={ cn('bl-customComponent-creditCard', classList) } style={ styles }>
+      <CardPreview
+        cardNumber={ cardNumber }
+        expiry={ expiry }
+        cvc={ cvc }
+        focused={ focus }
+        name={ cardholderName }
+        visibility={ cardPreviewVisibility }
```

Comment:
Wouldn't it be better if we check the visibility here, instead of passing cardPreviewVisibility through props to the CardPreview component!?

Like here:
``` 
{ cardPreviewVisibility && 
   <CardPreview
       cardNumber={ cardNumber }
       expiry={ expiry }
       cvc={ cvc }
       focused={ focus }
       name={ cardholderName }
       card={ card }
   />
} 
```

Reply:
For me it is more readable when we use the visibility property inside the component. Although I have nothing against your variant.
@v-excelsior what do you think?